src/python_testing/test_plan_table_generator.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env -S python3 -B
+#
+#    Copyright (c) 2024 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+import importlib
+import os
+from pathlib import Path
+
+import click
+from matter_testing_support import MatterTestConfig, generate_mobly_test_config
+
+
+def indent_multiline(multiline: str, num_spaces: int) -> str:
+    ''' Indents subsequent lines of a multiline string by num_spaces spaces'''
+    s = multiline.split('\n')
+    s = [(num_spaces * ' ' + line.lstrip()).rstrip() for line in s]
+    return '\n'.join(s).lstrip()
+
+
+@click.command()
+@click.argument('filename', type=click.Path(exists=True))
+@click.argument('classname', type=str)
+@click.argument('test', type=str)
+def main(filename, classname, test):
```

Comment:
Documentation about what this script does is too sparse. Suggest adding more documentation about what this tool does.

Reply:
Added desc - this will now print on the help menu.